unity3d/Assets/Swrve/SwrveSDK/SwrveNativeAndroid.cs

```diff
@@ -495,6 +499,12 @@ private string GetPushId(Dictionary<string, object> notification)
     public void OpenedFromPushNotification(string notificationJson)
     {
         Dictionary<string, object> notification = (Dictionary<string, object>)Json.Deserialize (notificationJson);
+        string gcmIdent = (string)notification[GCMIdentKey];
+        if(!string.IsNullOrEmpty(gcmIdent) && (gcmIdent == LastOpenedNotification)) {
+            return;
+        }
+        LastOpenedNotification = gcmIdent;
+
         string pushId = GetPushId(notification);
         SendPushEngagedEvent(pushId);
```

Comment:
Should that check only cover sending the engagement? At this point the push is opened and it won't do anything except open the app it we don't process the deeplink